### PR TITLE
Integrate Google Auth

### PR DESCRIPTION
**Pull request by winbrother**
- Allow google signin
- Allow google signout
- Add placeholder account

### CODE REVIEW

**Comment by winbrother** [comment]
image
image

File: data/model/user.tsx [hunk 1]
```diff
@@ -1,7 +1,12 @@
 export type User = {
-    id: number;
+    id: string;
     avatar?: string;
     name: string;
     dob: string;
     email?: string;
+    userType: UserType
```
Comment (lth08091998) [inline data/model/user.tsx:7]: should we include wallet address as well? btw what's dob?

File: components/login/LoginModal.tsx [hunk 2]
```diff
@@ -19,6 +23,29 @@ function LoginModal(props: ModalProps) {
...
 
+    const onGoogleLogin = (event: GoogleLoginResponse) => {
+        let profile = event.getBasicProfile();
+        let user : User = {
+            id: profile.getId(),
+            name: profile.getName(),
+            avatar: profile.getImageUrl(),
+            dob: 'abc',
+            email: profile.getEmail(),
+            userType: UserType.GOOGLE
```
Comment (lth08091998) [inline components/login/LoginModal.tsx:34]: Please check how to extract token id for our master server validation as well. Ref https://developers.google.com/identity/sign-in/web/backend-auth

File: data/model/user.tsx (hunk 1 above)
Comment (winbrother) [reply data/model/user.tsx:7]: this one is just placeholder, subject to change depending how master server handle Auth tokens.
Dob is date of birth btw

File: components/login/LoginModal.tsx (hunk 2 above)
Comment (winbrother) [reply components/login/LoginModal.tsx:34]: Basically, once login I will have to send token as Bearer to master server -> server will send that token to google apis to get the real Token and save it

**Review by lth08091998 — APPROVED** [review]

**Pull request merged**